cdragontoolbox/export.py

```diff
@@ -549,3 +551,28 @@ def convert_to_file(self, fin, fout):
         binfile = BinFile(fin)
         fout.write(json.dumps(binfile.to_serializable()).encode('ascii'))
 
+class SKNConverter(FileConverter):
+    def __init__(self, extensions):
+        self.extensions = extensions
+
+
+    def handle_path(self, path):
+        if self.extensions:
+            base, ext = os.path.splitext(path)
+            if ext in self.extensions:
+                return path
+        return None
+
+    def convert_to_file(self, fin, fout):
+        # close the writer and delete the extracted file, as each skn file has multiple output files
```

Comment:
It's important that actual output files are known and returned by `handle_path()`.

The current `FileConverter` does not support one file to multiple files conversions (because it was not needed).
If it is now needed it should be extended to support it properly. Note that current approach allowed to avoid recomputing the converted output path by storing it allong with the converter; this will need to be changed.

I'll think about it and try to find a good solution to this.

Reply:
Actually, since you put every file in a single directory, that might work without too much work. This would still require some changes to handle the fact that extracted "files" can now be directories (for instance, in `clean_output_dir()`).

Reply:
I'm leaving this, as I'm not sure what changes you'd want.

Reply:
If/when we end up exporting WPK and BNK files (voicelines and sfx), they also have multiple output files

Reply:
Would this be multiples files grouped in a directory or multiple files in the same (parent) directory (as it is for skn)?

Reply:
they should act the same as skn files with multiple models.